protected/views/dataset/view.php

```diff
@@ -746,3 +748,40 @@ class="btn btn-green btn-block claim-button"
 
     });
 </script>
+<script>
+    function goToPage() {
+        var targetPageNumber = document.getElementById('pageNumber').value;
+        var pageID = <?php echo $model->identifier?>;
+        //To validate page number
+        var userInput = parseInt(targetPageNumber);
+        var max = <?php echo $files->getDataProvider()->getPagination()->getPageCount() ?>;
+        //To output total pages
+        // console.log(max);
+        var min = 1;
+        if (userInput >= min && userInput <= max) {
+            console.log("Valid page number!");
+        }else if (userInput > max) {
+            targetPageNumber = max;
+            console.log("Error, return to " + max);
+        } else if (userInput < min) {
+            targetPageNumber = min;
+            console.log("Error, return to " + min);
+        }
+        // var targetUrlArray = Array.apply(null, Array(5)).map(function(_,i) { return window.location.pathname.split("/")[i]});]
+        // Create array with default values
+        //TODO: This array is hardcoded
+        let targetUrlArray = ["", "dataset", "view", "id", pageID];
+        targetUrlArray.push('Files_page', targetPageNumber);
+        window.location = window.location.origin + targetUrlArray.join("/");
+        // Uncomment will show the target url in console.
+        // console.log(window.location.origin + targetUrlArray.join("/"))
+    }
+
+    function pressEnter() {
```

Comment:
Is this JS function required?

Reply:
The JS function is removed.

Reply:
The `pressEnter` JS function is still present.

Reply:
The `pressEnter` is required as it returns `goToPage()`, then user can to go a specific page of a `file table` by inputing the page number and hitting the physical `Return` key.

Reply:
Ok now I understand that this function is responsible for detecting an Enter key press in the `pageNumber` text field and will then call the `goToPage()` JS function.

Reply:
Maybe calling this function `detectEnterKeyPress` is more accurate?

Reply:
This function name has been updated.